packages/common/src/visualizations/CartesianChartDataModel.ts

```diff
@@ -82,23 +83,81 @@ export class CartesianChartDataModel {
         const dimensions = this.resultsRunner.getDimensions();
         const metrics = this.resultsRunner.getMetrics();
 
-        const xColumn = dimensions[0];
-        const yColumn = metrics[0];
+        // TODO: the types could be cleaned up here to have fewer 'in' checks. 
+        // This works for 
```

Comment:
These types can be cleaned up, but this makes it work pretty well on both pages.